### ISSUE TITLE
Passing \noexpand\macroname to \pkg@warning fails

### ISSUE THREAD

**Issue by godbyk** [wspr/hardwrap#7]
The following generates an error:

```
\GeneratePackageLogMacros{mypkg}
\mypkg@warning{Testing \noexpand\textbf.}% should output 'Testing \textbf.'
```

Some existing packages use \noexpand\macroname to achieve the effect of printing '\macroname' verbatim to the log file.

**Comment by wspr** [comment]
I can't reproduce this. I get:

<pre>
Package mypkg Warning: Testing \textbf .
This message occurred on input line 5.
</pre>

How could this be?

**Comment by godbyk** [comment]
The example I had fail was `\noexpand\section`.

**Comment by wspr** [comment]
fix \noexpand usage (I hope)

closed by 3c6c5d01830c6372942e4f2d3822d14720eb3b80
